bindings/python/test/utils.py

```diff
@@ -54,3 +65,186 @@ def succeeded(self, event):
+        str: string(),
+        int: int64(),
+        float: float64(),
+        datetime.datetime: timestamp("ms"),
+        ObjectId: ObjectIdType(),
+        Decimal128: Decimal128StringType(),
+        bool: bool_(),
+    }
+
+    pytype_writeback_exc_map = {
+        str: None,
+        int: None,
+        float: None,
+        datetime.datetime: None,
+        ObjectId: pyarrow.lib.ArrowInvalid,
+        Decimal128: pyarrow.lib.ArrowInvalid,
+        bool: None,
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        if cls is TestNullsBase:
+            raise unittest.SkipTest("Base class")
+
+        if not client_context.connected:
+            raise unittest.SkipTest("cannot connect to MongoDB")
+        cls.cmd_listener = AllowListEventListener("find", "aggregate")
+        cls.getmore_listener = AllowListEventListener("getMore")
+        cls.client = client_context.get_client(
+            event_listeners=[cls.getmore_listener, cls.cmd_listener]
+        )
+
+        cls.oids = [ObjectId() for _ in range(4)]
+        cls.coll = cls.client.pymongoarrow_test.get_collection(
+            "test", write_concern=WriteConcern(w="majority")
+        )
+
+    def setUp(self):
+        self.coll.drop()
+
+        self.cmd_listener.reset()
+        self.getmore_listener.reset()
+
+    def assertType(self, obj1, arrow_type):
+        if isinstance(obj1, pyarrow.ChunkedArray):
+            if "storage_type" in dir(arrow_type):
+                self.assertEqual(obj1.type, arrow_type.storage_type)
+            else:
+                self.assertEqual(obj1.type, arrow_type)
+        else:
+            if isinstance(arrow_type, list):
+                self.assertTrue(obj1.dtype.name in arrow_type)
+            else:
+                self.assertEqual(obj1.dtype.name, arrow_type)
+
+    def test_int_handling(self):
+        # Default integral types
+        int_schema = Schema({"_id": ObjectIdType(), "int64": int64()})
+        int64_arr = [(i if (i % 2 == 0) else None) for i in range(len(self.oids))]
+        self.coll.insert_many(
+            [{"_id": self.oids[i], "int64": int64_arr[i]} for i in range(len(self.oids))]
+        )
+
+        table = self.find_fn(self.coll, {}, schema=int_schema)
+
+        # Resulting datatype should be float64 according to the spec for numpy
+        # and pandas.
+        atype = self.pytype_tab_map[int]
+        self.assertType(table["int64"], atype)
+
+        # Does it contain NAs where we expect?
+        self.assertTrue(np.all(np.equal(isna(int64_arr), isna(table["int64"]))))
+
+        # Write
+        self.coll.drop()
+        table_write = self.table_from_dict({"int64": int64_arr})
+
+        write(self.coll, table_write)
+        res_table = self.find_fn(self.coll, {}, schema=int_schema)
+
+        self.equal_fn(res_table["int64"], table_write["int64"])
+        self.assert_in_idx(res_table, "_id")
+        self.assertType(res_table["int64"], atype)
+
+    def test_other_handling(self):
```

Comment:
Can we add a test that will automatically fail when we add support for new data types? Something like:
```
def test_all_types():
    for type in _BsonArrowTypes:
        self.assertIn(type, self.pytype_tab_map)
```